WordPress/WordPressTest/Menus/MenuItemTests.swift

```diff
@@ -0,0 +1,150 @@
+
+        let fIsDescendantOfA = itemF.isDescendant(of: itemA)
+        XCTAssertFalse(fIsDescendantOfA)
+    }
+
+    /// Tests that the last descendant of an item is found.
+    func testLastDescendantInOrderedItems() {
+        let itemA = newMenuItem(named: "Item A")
+        let itemB = newMenuItem(named: "Item B")
+        let itemC = newMenuItem(named: "Item C")
+        let itemD = newMenuItem(named: "Item D")
+        let itemE = newMenuItem(named: "Item E")
+        let itemF = newMenuItem(named: "Item F")
+
+        let orderedItems = NSOrderedSet(array: [
+            itemA,
+            itemB,
+            itemC,
+            itemD,
+            itemE,
+            itemF
+        ])
+
+        /*
+         Item A
+         -- Item B
+         ---- Item C
+         Item D
+         -- Item E
+         -- Item F
+         */
+
+        itemB.parent = itemA
+        itemC.parent = itemB
+        itemE.parent = itemD
+        itemF.parent = itemD
+
+        let lastDescendant = itemA.lastDescendant(inOrderedItems: orderedItems)
+        XCTAssertEqual(lastDescendant, itemB)
+
+        let descendantOfItemC = itemC.lastDescendant(inOrderedItems: orderedItems)
+        XCTAssertEqual(descendantOfItemC, nil)
+
+        let descendantOfItemD = itemD.lastDescendant(inOrderedItems: orderedItems)
+        XCTAssertEqual(descendantOfItemD, itemF)
+    }
+
+    /// Tests that preceding siblings are found.
+    func testPrecedingSiblingInOrderedItems() {
+        let itemA = newMenuItem(named: "Item A")
+        let itemB = newMenuItem(named: "Item B")
+        let itemC = newMenuItem(named: "Item C")
+        let itemD = newMenuItem(named: "Item D")
+        let itemE = newMenuItem(named: "Item E")
+        let itemF = newMenuItem(named: "Item F")
+
+        let orderedItems = NSOrderedSet(array: [
+            itemA,
+            itemB,
+            itemC,
+            itemD,
+            itemE,
+            itemF
+        ])
+
+        /*
+         Item A
+         -- Item B
+         ---- Item C
+         Item D
+         -- Item E
+         -- Item F
+         */
+
+        itemB.parent = itemA
+        itemC.parent = itemB
+        itemE.parent = itemD
+        itemF.parent = itemD
+
+        let precedingSiblingForItemA = itemA.precedingSibling(inOrderedItems: orderedItems)
+        XCTAssertNil(precedingSiblingForItemA)
+
+        let precedingSiblingForItemD = itemD.precedingSibling(inOrderedItems: orderedItems)
+        XCTAssertEqual(precedingSiblingForItemD, itemA)
+
+        let precedingSiblingForItemF = itemF.precedingSibling(inOrderedItems: orderedItems)
+        XCTAssertEqual(precedingSiblingForItemF, itemE)
+    }
+
+    // MARK: - Private Helpers
+
+    fileprivate func newMenuItem(named name: String) -> MenuItem {
+        let entityName = MenuItem.classNameWithoutNamespaces()
+        let entity = NSEntityDescription.insertNewObject(forEntityName: entityName, into: context)
+
+        let menuItem = entity as! MenuItem
```

Comment:
`XCTUnwrap(entity as? MenuItem)` could be used here, but I'm not sure if it fits this context since this is a helper method and if the force-cast failed, it would be due to a programming error rather than a test failure.

Reply:
That's a good thought @guarani! Since the type shouldn't change and this is in the context of a test, I'm considering letting it stay as is. I believe this could only fail if some really bad stuff happened under the hood with Core Data.

Reply:
Sounds good!